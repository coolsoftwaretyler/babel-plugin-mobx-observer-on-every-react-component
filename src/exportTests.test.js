import { expect, test, describe } from "bun:test";
import { transform } from "@babel/core";
import pluginSyntaxJsx from "@babel/plugin-syntax-jsx";
import pluginTransReactJsx from "@babel/plugin-transform-react-jsx";
import autoObserverPlugin from "./index.ts";

const runTransform = (input) => {
    return transform(input, {
        plugins: [pluginSyntaxJsx, pluginTransReactJsx, autoObserverPlugin],
    });
};

describe("Mobx Observer Babel Plugin", () => {
    describe("function expression default export", () => {
        test("it wraps the component in observer", () => {
            const source = `
      
      import React from 'react';

function FunctionDeclaration() {
  return (
    <div>
      <h1>Function Declaration Component</h1>
      <p>This is an example of a React component using a function declaration.</p>
    </div>
  );
}

export default FunctionDeclaration;
      `
            const out = runTransform(source);

            expect(out.code).toMatchSnapshot();
        });
    });
});
