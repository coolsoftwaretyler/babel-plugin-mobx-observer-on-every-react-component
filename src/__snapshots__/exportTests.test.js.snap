// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Mobx Observer Babel Plugin function expression default export it wraps the component in observer 1`] = `
"import { observer } from "mobx-react";
import React from 'react';
const FunctionDeclaration = observer(function FunctionDeclaration() {
  return /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("h1", null, "Function Declaration Component"), /*#__PURE__*/React.createElement("p", null, "This is an example of a React component using a function declaration."));
});
export default FunctionDeclaration;"
`;
